external-crates/move/crates/move-analyzer/src/completion.rs

```diff
@@ -212,15 +220,144 @@ fn context_specific_lbrace(
     completions
 }
 
+fn fun_def_info(
+    symbols: &Symbols,
+    fhash: FileHash,
+    mod_ident: ModuleIdent_,
+    name: Symbol,
+) -> Option<&DefInfo> {
+    let Some(mdef) = symbols.mod_defs(&fhash, mod_ident) else {
+        return None;
+    };
+    let Some(fdef) = mdef.functions.get(&name) else {
+        return None;
+    };
+    let def_loc = DefLoc::new(fhash, fdef.start);
+    symbols.def_info(&def_loc)
+}
+
+fn fun_completion_item(
+    mod_ident: &ModuleIdent_,
+    method_name: &Symbol,
+    function_name: &Symbol,
+    type_args: &[Type],
+    arg_names: &[Symbol],
+    arg_types: &[Type],
+    ret_type: &Type,
+) -> CompletionItem {
+    let sig_string = format!(
+        "fun {}({}){}",
+        type_args_to_ide_string(type_args, /* verbose */ false),
+        type_list_to_ide_string(arg_types, /* verbose */ false),
+        ret_type_to_ide_str(ret_type, /* verbose */ false)
+    );
+    // we omit the first argument which is guaranteed to be there
+    // as this is a method and needs a receiver
+    let arg_snippet = arg_names[1..]
+        .iter()
+        .enumerate()
+        .map(|(idx, name)| format!("${{{}:{}}}", idx + 1, name))
+        .collect::<Vec<_>>()
+        .join(", ");
+    let label_details = Some(CompletionItemLabelDetails {
+        detail: Some(format!(
+            " ({}::{})",
+            mod_ident_to_ide_string(mod_ident),
+            function_name
+        )),
+        description: Some(sig_string),
+    });
+
+    CompletionItem {
+        label: format!("{}()", method_name,),
+        label_details,
+        kind: Some(CompletionItemKind::SNIPPET),
+        insert_text: Some(format!("{}({})", method_name, arg_snippet)),
+        insert_text_format: Some(InsertTextFormat::SNIPPET),
+        ..Default::default()
+    }
+}
+
+/// Handle dot auto-completion at a given position.
+fn dot(symbols: &Symbols, use_fpath: &Path, position: &Position) -> Vec<CompletionItem> {
```

Comment:
Two small nits about this function:

1. You should sort the methods by method name.
2. The fields should be not be sorted (they currently retain definition order), but should appear before the methods.

Reply:
As it appears, VSCode is doing the ordering of these for us (e.g., I can't reorder methods and fields). That being said, I added sorting of the method names (on the compiler side) for other clients